### PR TITLE
add support for `index_title` property in glossary pages

### PR DESCRIPTION
**Pull request by argshook**
This PR adds support for `index_title` on `pages/glossary/*.md` files

The questdb.io/glossary index will be generated as follows:

1. If `index_title` is defined, it will be used as the title
2. otherwise, the title will be generated from the filename:
    1. the filename will be split on `-`
    2. first characters of each word will be capitalized

thus, when `pages/glossary/time-series-database.md`:

* has `index_title: Time-Series Database`, the title will be `Time-Series Database`
* does not have `index_title`, the title will be `Time Series Database`

### CODE REVIEW

File: src/pages/glossary/time-series-database.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ---
-title: Time-Series Database
+index_title: Time-Series Database
+title: What Is a Time-Series Database?
```
Comment (amyshwang) [inline src/pages/glossary/time-series-database.md:3]: Does this mean that we no longer need to define the h1 title as it should be the same as the `title`?

**Review by amyshwang — COMMENTED** [review]
I will update the branch and get Database Partitioning fixed in this PR too.

File: src/pages/glossary/time-series-database.md (hunk 1 above)
Comment (argshook) [reply src/pages/glossary/time-series-database.md:3]: it's possible to do this way, but it's more work, unfortunately.

Perhaps for now we can have both?

Comment (amyshwang) [reply src/pages/glossary/time-series-database.md:3]: No - I removed h1 and the title is rendered on the page..

**Review by amyshwang — APPROVED** [review]
Looks good!

**Pull request merged**